go.mod

```diff
@@ -7,16 +7,17 @@ require (
 	github.com/aws/aws-sdk-go-v2/config v1.15.5
 	github.com/aws/aws-sdk-go-v2/service/s3 v1.26.9
 	github.com/dop251/goja v0.0.0-20220405120441-9037c2b61cbf
-	github.com/google/uuid v1.2.0
-	github.com/nspcc-dev/neo-go v0.98.2
-	github.com/nspcc-dev/neofs-sdk-go v1.0.0-rc.3.0.20220504192402-12ea1e8d740f
-	github.com/nspcc-dev/tzhash v1.5.2
-	github.com/stretchr/testify v1.7.1
+	github.com/google/uuid v1.3.0
+	github.com/nspcc-dev/neo-go v0.99.4
```

Comment:
> So, the root cause was the obsolete libs used in the client?

No, the only reason i updated the SDK lib (the others are just the consequences) is that it [panics](https://github.com/nspcc-dev/neofs-sdk-go/pull/319) when using `go1.19` (my case), so i decided to do that now since it will have been done someday anyway, and i did not want do any docker builds/downgrading to `go1.18`.

Basic error reasons i found: 
1. on my laptop removing objects was slower than writing them when using the same worker number, so finishing a test run led to keeping objects in the storage anyway;
2. i have seen some data races when some objects were tried to be removed twice (so not that optimal time utilization by the deleters) cause of bolt bucket cycling; 
3. on my laptop i found that it is impossible to be _totally_ in sync for deleters and writers: deleters do not know about ~200 objects (nonconstant value of course but quite common for my local runs) on every run. what i mean: the writers write their object IDs to the bolt but none of the readers are possible to see them until some time passes. Even if deleters are quite fast, some number of the objects are not available for them while the writers do their work (some sort of caching the outdated db snapshot? /cc @fyrchik)
4. No graceful tests shutdown: there is always a nonzero probability of keeping some objects in the nodes' storages by design in the current implementation.

I could not find any signs of a bug in a FSTree of the SN and could not reproduce that locally, so that is why I started changing the test's code but the PR contains too many changes already so I decided to make it iterativelly (mb you will add some implrevements too) and continue test runs.

Reply:
> I believe this is mitigated with the implementation of iteration that is based on channel, right?

I expect it to be so, yes.

> 1. Run k6 with just writers (without deleters), but with registry file.
> 2. Run k6 with just deleters (without writers) so that they delete objects created by writers.
> This way we rule out any race conditions between writers and deleters and can focus on whether object is deleted and disk space is released.

To be honest i just did not think about it :). I do not remember when exactly i stopped looking for a bug in a node and started reading the test (that was the first time i looked at the tests' code, i did not even know that rerunning tests with the same DB path is a _strange_ thing by design). Once again, i do not consider PR as a solution, just a little step to it. I faced some problems when tried to run tests and hope i made it better (IMO).